test/bokken_web/controllers/auth_controller_test.exs

```diff
@@ -0,0 +1,132 @@
+defmodule BokkenWeb.AuthControllerTest do
+  use BokkenWeb.ConnCase
+
+  import Bokken.Factory
+
+  setup %{conn: conn} do
+    {:ok, conn: put_req_header(conn, "accept", "application/json")}
+  end
+
+  describe "sign_up " do
+    test "sign_up new user when data is valid", %{conn: conn} do
+      user_params = %{
+        email: "example@mail.com",
+        password: "password1234",
+        role: "guardian"
+      }
+
+      conn = post(conn, ~p"/api/auth/sign_up", user_params)
+
+      assert json_response(conn, 201)
+    end
+
+    test "sign_up new user when data is invalid", %{conn: conn} do
+      user_params = %{
+        email: "example@mail.com",
+        password: "password1234",
+        role: "random"
+      }
+
+      conn = post(conn, ~p"/api/auth/sign_up", user_params)
+
+      assert json_response(conn, 422) == %{"errors" => %{"role" => ["não é válido"]}}
+    end
+  end
+
+  describe "sign_in " do
+    test "sign_in user when data is valid", %{conn: conn} do
+      user = insert(:user)
+
+      conn = post(conn, ~p"/api/auth/sign_in", %{email: user.email, password: user.password})
+
+      assert json_response(conn, 200)
+    end
+
+    test "sign_in user when data is invalid", %{conn: conn} do
+      user = insert(:user)
+
+      conn = post(conn, ~p"/api/auth/sign_in", %{email: user.email, password: "random1234"})
+
+      assert json_response(conn, 404) == %{"errors" => %{"detail" => "Not Found"}}
+    end
+  end
+
+  describe "show" do
+    test "shows current user when logged in", %{conn: conn} do
+      user = insert(:user)
+
+      conn = post(conn, ~p"/api/auth/sign_in", %{email: user.email, password: user.password})
+      conn = get(conn, ~p"/api/auth/me")
+
+      assert json_response(conn, 200)
+    end
+
+    test "throws error when not logged in", %{conn: conn} do
+      conn = get(conn, ~p"/api/auth/me")
+
+      assert json_response(conn, 401) == %{"error" => "unauthenticated"}
+    end
+  end
+
+  describe "sign_out" do
+    setup [:login_as_guardian]
+
+    test "sign_out user", %{conn: conn} do
+      insert(:user)
+
+      conn =
+        delete(conn, ~p"/api/auth/sign_out")
+        |> get(~p"/api/auth/me")
+
+      assert json_response(conn, 401) == %{"error" => "unauthenticated"}
+    end
+  end
+
+  describe "update" do
+    setup [:login_as_guardian]
+
+    test "update new user", %{conn: conn} do
+      user_params = %{
+        email: "random@gmail.com"
+      }
+
+      conn = put(conn, ~p"/api/auth/me", user: user_params)
+
+      assert json_response(conn, 200)
+    end
```

Comment:
In this test case, you should ensure that the user needs to confirm his email again. Maybe checking that `verified` is set to false?